validator/utils/synthetic/synthetic_generation_funcs.py

```diff
@@ -280,5 +285,6 @@ async def generate_synthetic_data(task: str) -> Any:
 
     func = getattr(sys.modules[__name__], generative_function_name)
     kwargs = task_config.synthetic_generation_config.kwargs
+    kwargs['redis_db'] = redis_db  # Add redis_db to the kwargs
```

Comment:
pretty obvious that is what is happening without the comment.

Comment could just explain why if you felt like we need one